pkg/pillar/dpcreconciler/linux.go

```diff
@@ -2050,6 +2171,7 @@ func (r *LinuxDpcReconciler) getIntendedMarkingRules(dpc types.DevicePortConfig,
 
 	if r.HVTypeKube {
 		protoMarkV4Rules = append(protoMarkV4Rules, defaultKubernetesIptablesRules()...)
```

Comment:
Seems like we have lot's of complexity for Kubernetes only for connection marking used by flow logging, which by default is not enabled anymore. Would it be possible to simplify all these rules somehow?
Also I'm surprised that nothing was added here for services: https://github.com/lf-edge/eve/blob/master/pkg/pillar/dpcreconciler/linux.go#L1825
Is service traffic only forwarded and not matched by INPUT rules? Should we define some filter/FORWARD rules for Kubernetes? Previously, forwarding was only used for apps, hence all the forwarding rules were handled by zedrouter microservice. Should nim take care of Kubernetes traffic forwarding filter rules?

Reply:
I think the first time when implemented this, I place those in both marking and input rules. but later I experimented removing one of them, and found the marking one could not be removed, but the input one after removing still working. the one simplification i can think of is remove the identifying the ip address, only to handle the service or whatever  ports. I thought if we do have a LB ip address we can use, it's better to add that. I can also remove that.

Reply:
@milan-zededa you are right. I had the 'flowlog' enabled on my projects, that's why the marking was working. i'm adding the HvTypeKube in the check.

Reply:
@milan-zededa actually this is not it allows the outside to come in through input. The reason we don't need to handle at those input is because Kubernetes dynamically handles the service port explicitly through the 'kube-proxy'. Each of the NodePort types or LB operated port is added/removed in the NAT table. That is the reason we only need to add/remove the marking rules in the mangle table.

Reply:
I see, so basically the [iptables flow](https://www.linkedin.com/pulse/iptables-commonly-used-rules-other-stories-short-once-zamani-rad/) for services is:
```
NAT Pre-routing (kube-proxy rules perform DNAT) -> (new dst not for this host) -> Mangle Forward -> etc. 
```

Reply:
Could you please add this explanation as a comment so that we do not forget why there are no input rules for Kubernetes services?

Reply:
will do.

Reply:
comments updated.

Reply:
Thanks, marking discussion as resolved.